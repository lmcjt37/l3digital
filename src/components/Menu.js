import React from 'react'
import Link from 'gatsby-link'

const Menu = props => (
  <nav id="menu">
    <div className="inner">
      <ul className="links">
        <li>
          <Link onClick={props.onToggleMenu} to="/">
            Home
          </Link>
        </li>
        <li>
          <Link onClick={props.onToggleMenu} to="/projects">
            Projects
          </Link>
        </li>
        <li>
          <Link onClick={props.onToggleMenu} to="/about">
            About Us
          </Link>
        </li>
      </ul>
    </div>
    <a className="close" onClick={props.onToggleMenu} href="javascript:;">
      Close
    </a>
  </nav>
)

Menu.propTypes = {
  onToggleMenu: React.PropTypes.func,
}

export default Menu
