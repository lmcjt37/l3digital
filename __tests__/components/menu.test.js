import Menu from '../../src/components/Menu'

describe('Menu tests', () => {
    test('check it renders correctly with snapshot', () => {
        const tree = render(<Menu onToggleMenu={jest.fn()} />)
        expect(tree).toMatchSnapshot()
    })
})
