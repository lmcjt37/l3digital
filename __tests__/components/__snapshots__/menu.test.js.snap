// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu tests check it renders correctly with snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        id="menu"
      >
        <div
          class="inner"
        >
          <ul
            class="links"
          >
            <li>
              <a
                href="/"
                state="[object Object]"
              >
                Home
              </a>
            </li>
            <li>
              <a
                href="/projects"
                state="[object Object]"
              >
                Projects
              </a>
            </li>
            <li>
              <a
                href="/about"
                state="[object Object]"
              >
                About Us
              </a>
            </li>
          </ul>
          <ul
            class="icons"
          >
            <li>
              <a
                class="icon alt fa-twitter"
                href="https://twitter.com/l3digital_uk"
                rel="noopener noreferrer"
                target="_blank"
              >
                <span
                  class="label"
                >
                  Twitter
                </span>
              </a>
            </li>
            <li>
              <a
                class="icon alt fa-github"
                href="https://github.com/l3digitaluk"
                rel="noopener noreferrer"
                target="_blank"
              >
                <span
                  class="label"
                >
                  GitHub
                </span>
              </a>
            </li>
            <li>
              <a
                class="icon alt fa-linkedin"
                href="https://www.linkedin.com/in/l3-digital-uk/"
                rel="noopener noreferrer"
                target="_blank"
              >
                <span
                  class="label"
                >
                  LinkedIn
                </span>
              </a>
            </li>
          </ul>
        </div>
        <a
          class="close"
          href="javascript:;"
        >
          Close
        </a>
      </nav>
    </div>
  </body>,
  "container": <div>
    <nav
      id="menu"
    >
      <div
        class="inner"
      >
        <ul
          class="links"
        >
          <li>
            <a
              href="/"
              state="[object Object]"
            >
              Home
            </a>
          </li>
          <li>
            <a
              href="/projects"
              state="[object Object]"
            >
              Projects
            </a>
          </li>
          <li>
            <a
              href="/about"
              state="[object Object]"
            >
              About Us
            </a>
          </li>
        </ul>
        <ul
          class="icons"
        >
          <li>
            <a
              class="icon alt fa-twitter"
              href="https://twitter.com/l3digital_uk"
              rel="noopener noreferrer"
              target="_blank"
            >
              <span
                class="label"
              >
                Twitter
              </span>
            </a>
          </li>
          <li>
            <a
              class="icon alt fa-github"
              href="https://github.com/l3digitaluk"
              rel="noopener noreferrer"
              target="_blank"
            >
              <span
                class="label"
              >
                GitHub
              </span>
            </a>
          </li>
          <li>
            <a
              class="icon alt fa-linkedin"
              href="https://www.linkedin.com/in/l3-digital-uk/"
              rel="noopener noreferrer"
              target="_blank"
            >
              <span
                class="label"
              >
                LinkedIn
              </span>
            </a>
          </li>
        </ul>
      </div>
      <a
        class="close"
        href="javascript:;"
      >
        Close
      </a>
    </nav>
  </div>,
  "debug": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllBySelectText": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getAllByValue": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getBySelectText": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "getByValue": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllBySelectText": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryAllByValue": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryBySelectText": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "queryByValue": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
